File: domain/src/main/java/com/thinkzi/weather/domain/usecase/base/SingleObserver.java
package com.thinkzi.weather.domain.usecase.base;

import io.reactivex.observers.DisposableSingleObserver;

/**
 * provide a observer for the work result (returned only T object or error)
 * */
public class SingleObserver<T> extends DisposableSingleObserver<T> {

    @Override
    public void onSuccess(T t) {

    }

    @Override
    public void onError(Throwable e) {

    }

}

File: domain/src/main/java/com/thinkzi/weather/domain/usecase/base/SingleObserver.kt
package com.thinkzi.weather.domain.usecase.base

import io.reactivex.observers.DisposableSingleObserver

/**
 * provide a observer for the work result (returned only T object or error)
 */
open class SingleObserver<T> : DisposableSingleObserver<T>() {

    override fun onSuccess(t: T) {

    }

    override fun onError(e: Throwable) {

    }

}
